src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -287,35 +286,45 @@ object JoinIndexRule
       l: LogicalPlan,
       r: LogicalPlan,
       condition: Expression): Boolean = {
-
+    val conditions = extractConditions(condition)
     // Output attributes from base relations. Join condition attributes must belong to these
-    // attributes
-    val lBaseAttrs = l.collectLeaves().filter(_.isInstanceOf[LogicalRelation]).flatMap(_.output)
-    val rBaseAttrs = r.collectLeaves().filter(_.isInstanceOf[LogicalRelation]).flatMap(_.output)
+    // attributes. We work on canonicalized forms to make sure we support case-sensitivity.
+    val lBaseAttrs = l
```

Comment:
nit: create a nested `def baseAttrs(p: LogicalPlan):`?

Reply:
hmm. looks like something similar is being used below. Maybe we can create
```scala
def relationOutputs(l: logicalPlan): ??? = {
  l.collectLeaves().filter(_.isInstanceOf[LogicalRelation]).flatMap(_.output)
}
```
so that this can be reused?

Reply:
thanks @imback82, changed as suggested